proto/buf/alpha/registry/v1alpha1/doc.proto

```diff
@@ -377,6 +378,47 @@ message FieldOptions {
   int32 jstype = 4;
 }
 
+// FieldLiteral represents a field and its value. It can be used to model descriptor options.
+message FieldLiteral {
+  // When is_extension is true, name is the fully-qualified name of the
+  // extension. Otherwise, it is the simple name of the field.
+  string name = 1;
+  // The unique field number associated with the field.
+  int32 tag = 2;
+  // Indicates whether this is an extension or not.
+  bool is_extension = 3;
+  // The field's value.
+  LiteralValue value = 4;
+}
+
+// MessageLiteral represents an option value for a type that is a message.
+message MessageLiteral {
+  // Map of field names to value. Extension field names will be in the form:
+  // [fully.qualified.extension.Name].
+  repeated FieldLiteral fields = 1;
+}
+
+// ArrayLiteral represents an option value for a type that is a repeated field.
+// Maps are also represented using an ArrayLiteral, where each map entry is a MessageLiteral
+// that contains a key and value field.
+message ArrayLiteral {
+  repeated LiteralValue elements = 1;
+}
+
+// LiteralValue represents the actual value of the extension.
+message LiteralValue {
+  oneof value {
+    string string_value = 2;
+    int64 int_value = 3;
+    uint64 uint_value = 4;
+    double double_value = 5;
+    bool bool_value = 6;
+    string enum_value = 7;
```

Comment:
Why is this `string`?

Reply:
We going to be showing this in docs so the name should be more useful than the number value to users.

Reply:
@jhump should this semantically be ie `enum_value_name_value`, if that makes sense? It's not an `enum`, its an enum value, and its the name of the enum value...I don't know.

Reply:
That seems like a lot of ceremony. It represents an enum value, so seems like a doc comment clarifying that this is the enum value name feels like it suffices. Another option is to extract out another type -- like maybe an `EnumLiteral` that has both a name and a number. But, honestly, that felts a little overkill 🤷